### PR TITLE
Fix find-definition stack not working in upcoming emacs versions

### LINKED ISSUE #699: `slime-pop-find-definition-stack' not working on recent emacs master
Because `find-tag-marker-ring' is obsolete since 25.1 inserting the point marker on M-. does nothing anymore.

### PR DESCRIPTION
**Pull request by astrangeguy**
This fixes #699

### CODE REVIEW

File: slime.el [hunk 1]
```diff
@@ -3672,14 +3672,14 @@ alist but ignores CDRs."
...
-  (ring-insert find-tag-marker-ring (point-marker)))
+  "Add point to find-tag-marker-stack."
+  (require 'xref)
+  (xref-push-marker-stack (point-marker)))
 
 (defun slime-pop-find-definition-stack ()
   "Pop the edit-definition stack and goto the location."
   (interactive)
-  (pop-tag-mark))
+  (xref-go-back))
```
Comment (luismbo) [inline slime.el:3682]: `xref-go-back` doesn't exist in, e.g., Emacs 27.1

Comment (luismbo) [reply slime.el:3682]: I guess we should use `xref-pop-marker-stack` instead?

Comment (astrangeguy) [reply slime.el:3682]: Yes, stupid of me not to notice the old name 3 lines above...

File: slime.el [hunk 2]
```diff
@@ -3672,14 +3672,14 @@ alist but ignores CDRs."
 ;;;; Edit definition
 
 (defun slime-push-definition-stack ()
-  "Add point to find-tag-marker-ring."
-  (require 'etags)
-  (ring-insert find-tag-marker-ring (point-marker)))
+  "Add point to find-tag-marker-stack."
+  (require 'xref)
```
Comment (luismbo) [inline slime.el:3676]: While we’re at it. Why require at runtime? Can’t we just move this dependency to the beginning of the file like every other dependency?

Comment (astrangeguy) [reply slime.el]: Seems to be an artifact of something from the past... the runtime-require of etags is unnecessary since it's a dependency ether way

Comment (astrangeguy) [reply slime.el]: Yeah it's from when etags was loaded only when needed on XEmacs

**Comment by luismbo** [comment]
@astrangeguy great, thanks. 👍

**Pull request merged**